nova/tests/unit/objects/test_console_auth_token.py

```diff
@@ -75,21 +80,23 @@ def _test_authorize(self, console_type, mock_create):
 
         url = obj.access_url
         if console_type != 'novnc':
-            expected_url = '%s?token=%s' % (
-                fakes.fake_token_dict['access_url_base'],
-                fakes.fake_token)
+            expected_url = '%s?token=%s' % (base_url, fakes.fake_token)
         else:
-            path = urlparse.urlencode({'path': '?token=%s' % fakes.fake_token})
-            expected_url = '%s?%s' % (
-                fakes.fake_token_dict['access_url_base'],
-                path)
+            if 'path=' in base_url:
+                expected_url = base_url + urlparse.quote('?token=%s' % token)
+            else:
+                path = urlparse.urlencode({'path': '?token=%s' %
+                                                   fakes.fake_token})
+                expected_url = '%s?%s' % (base_url, path)
         self.assertEqual(expected_url, url)
 
     def test_authorize(self):
         self._test_authorize(fakes.fake_token_dict['console_type'])
 
     def test_authorize_novnc(self):
         self._test_authorize('novnc')
+        self._test_authorize('novnc', base_url=fakes.fake_token_dict[
+            'access_url_base'] + '?path=fake-path')
```

Comment:
For upsteraming this patch, I think we should move this line into its own testfunction called something along the lines of `test_authorize_novnc_with_path_in_base_url`.

Reply:
Done.

Reply:
Thanks. Since we don't have `pep8` tests, I'd like you to move the `base_url` into a variable for this to look nicer. Building that parameter over 2 lines doesn't look too nice.